main_test.go

```diff
@@ -0,0 +1,1986 @@
+	"os"
+	"os/exec"
+	"os/user"
+	"path/filepath"
+	"sort"
+	"strings"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/VertebrateResequencing/wr/jobqueue"
+	"github.com/VertebrateResequencing/wr/jobqueue/scheduler"
+	. "github.com/smartystreets/goconvey/convey"
+	"github.com/wtsi-ssg/wrstat/v4/basedirs"
+	"github.com/wtsi-ssg/wrstat/v4/dgut"
+	"github.com/wtsi-ssg/wrstat/v4/internal/fixtimes"
+	"github.com/wtsi-ssg/wrstat/v4/summary"
+)
+
+const app = "wrstat_test"
+
+func buildSelf() func() {
+	cmd := exec.Command(
+		"go", "build", "-tags", "netgo",
+		"-ldflags=-X github.com/wtsi-ssg/wrstat/v4/cmd.runJobs=0 -X github.com/wtsi-ssg/wrstat/v4/cmd.Version=TESTVERSION",
+		"-o", app,
+	)
+
+	cmd.Env = append(os.Environ(), "CGO_ENABLED=1")
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	if err := cmd.Run(); err != nil {
+		failMainTest(err.Error())
+
+		return nil
+	}
+
+	return func() {
+		os.Remove(app)
+	}
+}
+
+func failMainTest(err string) {
+	fmt.Println(err) //nolint:forbidigo
+}
+
+func TestMain(m *testing.M) {
+	d1 := buildSelf()
+	if d1 == nil {
+		return
+	}
+
+	defer os.Exit(m.Run())
+	defer d1()
+}
+
+func runWRStat(args ...string) (string, string, []*jobqueue.Job, error) {
+	var (
+		stdout, stderr strings.Builder
+		jobs           []*jobqueue.Job
+	)
+
+	pr, pw, err := os.Pipe()
+	if err != nil {
+		return "", "", nil, err
+	}
+
+	cmd := exec.Command("./"+app, args...)
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	cmd.ExtraFiles = append(cmd.ExtraFiles, pw)
+
+	jd := json.NewDecoder(pr)
+	done := make(chan struct{})
+
+	go func() {
+		for {
+			var j []*jobqueue.Job
+
+			if errr := jd.Decode(&j); errr != nil {
+				break
+			}
+
+			jobs = append(jobs, j...)
+		}
+
+		close(done)
+	}()
+
+	err = cmd.Run()
+
+	pw.Close()
+
+	<-done
+
+	return stdout.String(), stderr.String(), jobs, err
+}
+
+func TestVersion(t *testing.T) {
+	Convey("wrstat prints the correct version", t, func() {
+		output, stderr, _, err := runWRStat("version")
+		So(err, ShouldBeNil)
+		So(strings.TrimSpace(output), ShouldEqual, "TESTVERSION")
+		So(stderr, ShouldBeBlank)
+	})
+}
+
+func TestCron(t *testing.T) {
+	Convey("For the cron subcommand", t, func() {
+		multiTests(t, "cron", "-c", "* * * * * *")
+	})
+}
+
+func multiTests(t *testing.T, subcommand ...string) {
+	t.Helper()
+
+	walkReqs := &scheduler.Requirements{
+		RAM:   16000,
+		Time:  19 * time.Hour,
+		Cores: 1,
+		Disk:  1,
+	}
+
+	combineReqs := &scheduler.Requirements{
+		RAM:   800,
+		Time:  40 * time.Minute,
+		Cores: 1,
+		Disk:  1,
+	}
+
+	touchReqs := &scheduler.Requirements{
+		RAM:   100,
+		Time:  10 * time.Second,
+		Cores: 1,
+		Disk:  1,
+	}
+
+	baseDirsReqs := &scheduler.Requirements{
+		RAM:   42000,
+		Time:  15 * time.Minute,
+		Cores: 1,
+		Disk:  1,
+	}
+
+	tidyReqs := &scheduler.Requirements{
+		RAM:   100,
+		Time:  10 * time.Second,
+		Cores: 1,
+		Disk:  1,
+	}
+
+	date := time.Now().Format("20060102")
+
+	Convey("wrstat gets the stats for a partial run", func() {
```

Comment:
These test descriptions seem misleading? They should reflect that we're just testing the correct commands would be run?